packages/react-components/src/components/MessageStatusIndicator.tsx

```diff
@@ -114,40 +109,23 @@ export const MessageStatusIndicator = (props: MessageStatusIndicatorProps): JSX.
         </TooltipHost>
       );
     case 'seen':
-      if (
-        remoteParticipantsCount &&
-        remoteParticipantsCount > 1 &&
-        message &&
-        readReceiptForEachSender &&
-        isToolTipHovered
-      ) {
-        const readBy: { id: string; name: string }[] = getParticipantsWhoHaveReadMessage(
-          message,
-          readReceiptForEachSender
-        );
-        messageThreadReadCount = readBy.length;
-      }
-
       return (
         <TooltipHost
           content={
             // when it's just 1 to 1 texting, we don't need to know who has read the message, just show message as 'seen'
             // when readcount is 0, we have a bug, show 'seen' to cover up as a fall back
             // when participant count is 0, we have a bug, show 'seen' to cover up as a fall back
-            messageThreadReadCount === 0 ||
+            readCount === 0 ||
             (remoteParticipantsCount && remoteParticipantsCount <= 1) ||
-            !messageThreadReadCount ||
+            !readCount ||
             !remoteParticipantsCount ||
             strings.readByTooltipText === undefined
               ? strings.seenTooltipText
               : _formatString(strings.readByTooltipText, {
-                  messageThreadReadCount: `${messageThreadReadCount}`,
+                  messageThreadReadCount: `${readCount}`,
```

Comment:
I can still see perf issue here, and I would say even readCount and remoteParticipant count parameter isn't necessary, expose a toolTipText and onHover function will make this component more generic